src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/Conversions.cs

```diff
@@ -41,6 +41,21 @@ protected override ConversionsBase WithNullabilityCore(bool includeNullability)
             return new Conversions(_binder, currentRecursionDepth, includeNullability, this);
         }
 
+        public static void CheckDefaultParameterMatch(ErrorCode code, ParameterSymbol source, ParameterSymbol target, int paramIdx, Location location, BindingDiagnosticBag diagnostics)
+        {
+            var sourceParamDefault = source.ExplicitDefaultConstantValue;
+
+            if (sourceParamDefault is not null && !sourceParamDefault.IsBad)
+            {
+                var targetParamDefault = target.ExplicitDefaultConstantValue;
+                if ((!targetParamDefault?.IsBad ?? true) && sourceParamDefault != targetParamDefault)
```

Comment:
The `!` and precendence is hurting my head here. Consider just using `targetParamDefault?.IsBad != false`